coordinator/src/main/scala/filodb.coordinator/NodeCoordinatorActor.scala

```diff
@@ -190,7 +191,11 @@ class NodeCoordinatorActor(metaStore: MetaStore,
       if(entries.length > 0 ) {
         entries.foreach { ingestion =>
           val data = ingestion.toString().split("\001")
-          val ref = DatasetRef(data(2), Some(data(1)))
+          val databaseOpt = if (data(1).isEmpty || data(1).equals("None")) None else Some(data(1))
+          var ref = DatasetRef(data(2), databaseOpt)
+          if(ref.database.getOrElse("") == "") {
```

Comment:
I don't understand why this if is needed.  When databaseOpt is None, that is exactly the same as doing `DatasetRef(data(2))`..  in other words this code doesn't do anything.